py-polars/polars/testing/asserts.py

```diff
@@ -75,24 +75,21 @@ def assert_frame_equal(
     >>> assert_frame_equal(df1, df2)  # doctest: +SKIP
     AssertionError: Values for column 'a' are different.
     """
-    if isinstance(left, LazyFrame) and isinstance(right, LazyFrame):
-        left, right = left.collect(), right.collect()
-        obj = "LazyFrames"
+    if collect_input_frames := (
+        isinstance(left, LazyFrame) and isinstance(right, LazyFrame)
+    ):
+        objs = "LazyFrames"
```

Comment:
This is not a nice usage of the walrus imo - we need the variable outside of this if statement anyway. So just write:
```suggestion
    collect_input_frames = isinstance(left, LazyFrame) and isinstance(right, LazyFrame)
    if collect_input_frames:
        objs = "LazyFrames"
```

Reply:
It actually _is_ scoped so it's visible beyond of the `if` - but I won't fight for this one lol...
Yours is clean and straightforward, and I can get a bit walrus-happy sometimes 🤣